utils/spdx/src/main/kotlin/SpdxExpression.kt

```diff
@@ -201,29 +201,41 @@ sealed class SpdxExpression {
     /**
      * Concatenate [this][SpdxExpression] and [other] using [SpdxOperator.AND].
      */
-    infix fun and(other: SpdxExpression) = SpdxCompoundExpression(this, SpdxOperator.AND, other)
+    infix fun and(other: SpdxExpression) = SpdxCompoundExpression(listOf(this, other), SpdxOperator.AND)
 
     /**
      * Concatenate [this][SpdxExpression] and [other] using [SpdxOperator.OR].
      */
-    infix fun or(other: SpdxExpression) = SpdxCompoundExpression(this, SpdxOperator.OR, other)
+    infix fun or(other: SpdxExpression) = SpdxCompoundExpression(listOf(this, other), SpdxOperator.OR)
 }
 
 /**
- * An SPDX expression compound of a [left] and a [right] expression with an [operator] as defined by version 2.1 of the
- * [SPDX specification, appendix IV][1].
+ * An SPDX expression compound of a list of [child][children] expressions with an [operator] as defined by version 2.1
+ * of the [SPDX specification, appendix IV][1].
  *
  * [1]: https://spdx.dev/spdx-specification-21-web-version#h.jxpfx0ykyb60
  */
-class SpdxCompoundExpression(
-    val left: SpdxExpression,
-    val operator: SpdxOperator,
-    val right: SpdxExpression
+class SpdxCompoundExpression internal constructor(
+    val children: List<SpdxExpression>,
+    val operator: SpdxOperator
```

Comment:
Sorry, one more question: Is the order of parameters intended like this? To me it seems like we should align to line 232 now and make the `operator` go first.